frontend/packages/dev-console/src/components/topology/actions/workloadActions.ts

```diff
@@ -0,0 +1,37 @@
+import * as _ from 'lodash';
+import { KebabAction, KebabOption } from '@console/internal/components/utils';
+import { modelFor } from '@console/internal/module/k8s';
+import { menuActions as deploymentConfigMenuActions } from '@console/internal/components/deployment-config';
+import { menuActions as deploymentMenuActions } from '@console/internal/components/deployment';
+import { menuActions as statefulSetMenuActions } from '@console/internal/components/stateful-set';
+import { menuActions as daemonSetMenuActions } from '@console/internal/components/daemon-set';
+import { TopologyDataObject } from '../topology-types';
+import { getResourceDeploymentObject } from '../topology-utils';
+
+export const WorkloadActions = (workload: TopologyDataObject): KebabOption[] => {
+  const contextMenuResource = getResourceDeploymentObject(workload);
+
+  if (!contextMenuResource) {
+    return null;
+  }
+
+  let menuActions: KebabAction[] = [];
+  switch (contextMenuResource.kind) {
+    case 'DeploymentConfig':
+      menuActions = deploymentConfigMenuActions;
+      break;
+    case 'Deployment':
+      menuActions = deploymentMenuActions;
+      break;
+    case 'StatefulSet':
+      menuActions = statefulSetMenuActions;
+      break;
+    case 'DaemonSet':
+      menuActions = daemonSetMenuActions;
+      break;
+    default:
+      menuActions = [];
```

Comment:
nit: no default case needed if initialized already to empty array. Or vise versa.

Reply:
Removed the initialization